lib/ohai/plugins/linux/sessions.rb

```diff
@@ -0,0 +1,55 @@
+#
+# Author:: Davide Cavalca <dcavalca@fb.com>
+# Copyright:: Copyright (c) 2016 Facebook
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "ohai/util/file_helper"
+
+include Ohai::Util::FileHelper
+
+Ohai.plugin(:Sessions) do
+  provides "sessions/by_session", "sessions/by_user"
+
+  collect_data(:linux) do
+    sessions Mash.new unless sessions
```

Comment:
The sessions mash creation should get moved down after the shellout so if we fail to shellout and gather the information we don't end up with an empty session mash sitting on the output.